src/transformers/onnx/__main__.py

```diff
@@ -65,48 +65,108 @@ def main():
     if not args.output.parent.exists():
         args.output.parent.mkdir(parents=True)
 
-    # Instantiate the appropriate preprocessor
-    if args.preprocessor == "auto":
-        preprocessor = get_preprocessor(args.model)
-    elif args.preprocessor == "tokenizer":
-        preprocessor = AutoTokenizer.from_pretrained(args.model)
-    elif args.preprocessor == "feature_extractor":
-        preprocessor = AutoFeatureExtractor.from_pretrained(args.model)
-    elif args.preprocessor == "processor":
-        preprocessor = AutoProcessor.from_pretrained(args.model)
-    else:
-        raise ValueError(f"Unknown preprocessor type '{args.preprocessor}'")
-
     # Allocate the model
     model = FeaturesManager.get_model_from_feature(
         args.feature, args.model, framework=args.framework, cache_dir=args.cache_dir
     )
+
     model_kind, model_onnx_config = FeaturesManager.check_supported_model_or_raise(model, feature=args.feature)
     onnx_config = model_onnx_config(model.config)
 
-    # Ensure the requested opset is sufficient
-    if args.opset is None:
-        args.opset = onnx_config.default_onnx_opset
+    if model_kind == "vision-encoder-decoder":
```

Comment:
This line create 2 workflows based on the type of the model. The first is for visionencoderdecoder and second is the old export workflow. This is currently hardcoded for 'vision-encoder-decoder'. Would appreciate feedback if we can do this better.

Another approach I was thinking is to make it generic like: model.hasattr(encoder) and model.hasattr(decoder). This should cover all models with two parts in config. But I am not sure of any problems that can occur in future.

Reply:
Yeah, ideally we would want something generic that also makes it easy to add the other encoder-decoder models like:

* https://huggingface.co/docs/transformers/v4.22.2/en/model_doc/encoder-decoder
* https://huggingface.co/docs/transformers/v4.22.2/en/model_doc/speech-encoder-decoder

Your idea to use `model.hasattr(encoder)` won't be sufficient because all seq2seq models like `t5` also have this attribute. I've asked internally to see if the core maintainers know how we can distinguish this case.

Internal Slack thread: https://huggingface.slack.com/archives/C01N44FJDHT/p1664797000511649

Reply:
Updated

Reply:
> This line create 2 workflows based on the type of the model. The first is for visionencoderdecoder and second is the old export workflow. This is currently hardcoded for 'vision-encoder-decoder'. Would appreciate feedback if we can do this better.
> 
> Another approach I was thinking is to make it generic like: model.hasattr(encoder) and model.hasattr(decoder). This should cover all models with two parts in config. But I am not sure of any problems that can occur in future.

Hello,when i convert trocr model to encoder.onnx and decoder.onnx,why we used "processor = TrOCRProcessor.from_pretrained("original basemodel")", can i use encoder.onnx or decoder.onnx replace it?

Reply:
Hi @dcdethan could you please elaborate more on the issue? Why would you like to use the onnx models to load the preprocessor?  Could you please create a new issue in `Optimum` as ONNX exports have been migrated to the exporters in `Optimum`. https://github.com/huggingface/optimum/issues

Reply:
> Hi @dcdethan could you please elaborate more on the issue? Why would you like to use the onnx models to load the preprocessor? Could you please create a new issue in `Optimum` as ONNX exports have been migrated to the exporters in `Optimum`. https://github.com/huggingface/optimum/issues

ok,i will add a issue as you said.I use your example, the link i [onnx_trocr_inference.py](https://gist.github.com/mht-sharma/f38c670930ac7df413c07327e692ee39#file-onnx_trocr_inference-py)